signal_processing/feature_extraction.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import neurokit2 as nk
+from math import log10
+from scipy import signal
+
+
+# Use pan tompkins algorithm to separate signal into n beats
+def apply_pan_tompkins(x, n_beats=110, fs=1000):
+    _, rpeaks = nk.ecg_peaks(x, sampling_rate=fs)
+
+    # Select 250 left of R and 500 right of R as one beat
+    cleaned_signal_dict = {}
+    for peak in rpeaks['ECG_R_Peaks']:
+        cleaned_signal_dict.update({peak: x[peak - 250:peak + 500]})
+
+    # If n_beats is defined, returned those beats
+    if n_beats:
```

Comment:
Won't this code always land inside the if since you are defining n_beats to be 110 in the function? Think it should be set to None as the default behaviour but up to you

Reply:
Leaving unchanged. Don't think it acc matters